app/static/js/initialize_map.js

```diff
@@ -0,0 +1,119 @@
+geocoder = new google.maps.Geocoder();
+address = document.getElementById("address");
+
+// Global array for map markers
+var markers = []
+
+// Global Marker Wrappers and Map
+var globalMarkerWrappers = null;
+var globalMap = null;
+
+// Initial map center coordinates
+INITIAL_CENTER_LAT = 39.952;
+INITIAL_CENTER_LONG = -75.195;
+initial_coords = new google.maps.LatLng(INITIAL_CENTER_LAT, INITIAL_CENTER_LONG);
+
+// Initial map zoom value
+ZOOM = 17;
+
+// Bounds for the Date Slider. End date is current date.
+var BOUNDS_MIN = new Date(2015, 0, 1);
+var BOUNDS_MAX = new Date();
+
+function MarkerWrapper(actualMarker, incidentDate, contentString) {
+    this.actualMarker = actualMarker;
+    this.incidentDate = incidentDate;
+    this.contentString = contentString;
+}
+
+function storeMarkerState(markerWrappers, map, minDate) {
+    globalMarkerWrappers = markerWrappers;
+    globalMap = map;
+    console.log(markerWrappers[0].actualMarker);
+    for (mw = 0; mw < globalMarkerWrappers.length; mw++)
+    {
+        (globalMarkerWrappers[mw].actualMarker).setMap(globalMap);
```

Comment:
Interesting...

I didn't know javascript had for-each loops, maybe they work strangely